tests/unit/contracts/test_standalone.py

```diff
@@ -0,0 +1,162 @@
+            "mypackage.green.bar",
+            "mypackage.yellow",
+            "mypackage.yellow.gamma",
+            "mypackage.yellow.delta",
+            "mypackage.other",
+            "mypackage.other.sub",
+            "mypackage.other.sub2",
+        ):
+            graph.add_module(module)
+        return graph
+
+    def _check_default_contract(self, graph):
+        contract = StandaloneContract(
+            name="Standalone contract",
+            session_options={"root_packages": ["mypackage"]},
+            contract_options={"modules": ("mypackage.green", "mypackage.yellow")},
+        )
+        return contract.check(graph=graph, verbose=False)
+
+    def test_when_modules_are_standalone(self):
+        graph = self._build_default_graph()
+        graph.add_import(
+            importer="mypackage.blue",
+            imported="mypackage.other",
+            line_number=10,
+            line_contents="-",
+        )
+        graph.add_import(
+            importer="mypackage.other",
+            imported="mypackage.blue.world",
+            line_number=11,
+            line_contents="-",
+        )
+
+        contract_check = self._check_default_contract(graph)
+
+        assert contract_check.kept, contract_check.metadata
+
+    def test_non_standalone_imported(self):
+        graph = self._build_default_graph()
+        graph.add_import(
+            importer="mypackage.blue",
+            imported="mypackage.green",
+            line_number=10,
+            line_contents="-",
+        )
+        contract_check = self._check_default_contract(graph)
+
+        assert not contract_check.kept
+
+        expected_metadata = {
+            "violations": {"mypackage.green": [("mypackage.green", "mypackage.blue")]}
+        }
+        assert expected_metadata == contract_check.metadata
+
+    def test_non_standalone_imports(self):
+        graph = self._build_default_graph()
+        graph.add_import(
+            importer="mypackage.yellow",
+            imported="mypackage.other",
+            line_number=10,
+            line_contents="-",
+        )
+
+        contract_check = self._check_default_contract(graph)
+
+        assert not contract_check.kept
+
+        expected_metadata = {
+            "violations": {"mypackage.yellow": [("mypackage.other", "mypackage.yellow")]}
+        }
+        assert expected_metadata == contract_check.metadata
+
+    def test_standalone_ignore(self):
+        graph = self._build_default_graph()
+        graph.add_import(
+            importer="mypackage.yellow",
+            imported="mypackage.other",
+            line_number=10,
+            line_contents="-",
+        )
+
+        contract = StandaloneContract(
+            name="Standalone contract",
+            session_options={"root_packages": ["mypackage"]},
+            contract_options={
+                "modules": ("mypackage.green", "mypackage.yellow"),
+                "ignore_imports": ["mypackage.yellow -> mypackage.other"],
```

Comment:
Very minor comment, but we could allow passing in the `ignore_imports` as an optional argument to the utility method.